app/services/workflow_link_service.rb

```diff
@@ -0,0 +1,28 @@
+class WorkflowLinkService
+  attr_accessor :workflow_id
+
+  def initialize(workflow_id)
+    self.workflow_id = workflow_id
+  end
+
+  def link(options)
+    TagLink.where(tag_link(options)).first_or_create
+    nil
+  end
+
+  def unlink(options)
+    TagLink.find_by(tag_link(options)).try(:destroy)
+  end
+
+  def find(options_arr)
+    options_arr.collect do |options|
+      TagLink.where(tag_link(options)).pluck(:workflow_id).first
+    end
+  end
+
+  private
+
+  def tag_link(options)
+    options.merge(:workflow_id => workflow_id)
```

Comment:
@bzwei Why is this merging the workflow_id, shouldn't it be using the object_type, app_name to do the match and then pickup the TagLink object.